Rick-and-Morty/Rick And Morty/ViewModels/CharactersViewModel.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+final class CharactersViewModel: ObservableObject {
+    let characterType: CharacterType
+    
+    @Published var characters: [Character] = []
+    var title: String = ""
+    var imagePosition: CharacterImagePosition = .right
+    
+    init(characterType: CharacterType) {
+        self.characterType = characterType
+        self.title = getTitle()
+        self.imagePosition = getImagePosition()
+        
+        fetchCharacters()
+    }
+    
+    private func fetchCharacters() {
+        if characterType == .rick {
+            characters = ricks
+        } else {
+            characters = morties
+        }
+    }
+    
+    private func getTitle() -> String {
+        if characterType == .rick {
+            return "Ricks"
+        } else {
+            return "Morties"
+        }
+    }
+    
+    private func getImagePosition() -> CharacterImagePosition {
+        if characterType == .rick {
+            return .right
+        } else {
+            return .left
+        }
+    }
+}
+
+extension CharactersViewModel {
+    enum CharacterType {
+        case rick
+        case morty
+    }
+}
```

Comment:
Interesting: this is violating the Open Closed principle :P 
You would have to update the ViewModel if you want to add `Summer` or `Jerry`

Why don't you define a struct instead? 

```
struct CharacterViewState {
    let title: String
    let imagePosition: CharacterImagePosition
}
```

If you feed your ViewModel with an array of this model, you won't need to take decisions anymore inside your ViewModel ;)

Reply:
I created a CharacterViewState and a CharactersViewState which holds an array of CharacterViewState's (naming could probably be better). 

`struct CharactersViewState: Identifiable {
    let id = UUID()
    let title: String
    let iconName: String
    let characterViewStates: [CharacterViewState]
}`

`struct CharacterViewState: Identifiable {
    let id = UUID()
    let name: String
    let description: String
    let shortDescription: String?
    let imageName: String
    let imagePosition: CharacterImagePosition
}`

A CharactersViewState is made for each CharacterType in the new ContentViewModel. I don't think there is a need for a CharactersViewModel anymore. Check out my new commit, it should make more sense.